pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -236,6 +206,48 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 	}
 
 	for _, replset := range repls {
+		if replset.Storage == nil {
+			replset.Storage = cr.Spec.Mongod.Storage
+		}
+
+		switch replset.Storage.Engine {
+		case StorageEngineInMemory:
+			if replset.Storage.InMemory == nil {
+				replset.Storage.InMemory = &MongodSpecInMemory{}
+			}
+			if replset.Storage.InMemory.EngineConfig == nil {
+				replset.Storage.InMemory.EngineConfig = &MongodSpecInMemoryEngineConfig{}
+			}
+			if replset.Storage.InMemory.EngineConfig.InMemorySizeRatio == 0 {
+				replset.Storage.InMemory.EngineConfig.InMemorySizeRatio = defaultInMemorySizeRatio
+			}
+		case StorageEngineWiredTiger:
+			if replset.Storage.WiredTiger == nil {
+				replset.Storage.WiredTiger = &MongodSpecWiredTiger{}
+			}
+			if replset.Storage.WiredTiger.CollectionConfig == nil {
+				replset.Storage.WiredTiger.CollectionConfig = &MongodSpecWiredTigerCollectionConfig{}
+			}
+			if replset.Storage.WiredTiger.EngineConfig == nil {
+				replset.Storage.WiredTiger.EngineConfig = &MongodSpecWiredTigerEngineConfig{}
+			}
+			if replset.Storage.WiredTiger.EngineConfig.CacheSizeRatio == 0 {
+				replset.Storage.WiredTiger.EngineConfig.CacheSizeRatio = defaultWiredTigerCacheSizeRatio
+			}
+			if replset.Storage.WiredTiger.IndexConfig == nil {
+				replset.Storage.WiredTiger.IndexConfig = &MongodSpecWiredTigerIndexConfig{
+					PrefixCompression: true,
+				}
+			}
+		}
+
+		if cr.Spec.Sharding.Enabled && replset.Storage.Engine == StorageEngineMMAPv1 {
```

Comment:
we don't support it at all
```suggestion
		if replset.Storage.Engine == StorageEngineMMAPv1 {
```

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/467/commits/7a642b9c66d5d6d42e70e5d8983f34bf461ad22d